doc/reference/usb/uds_cdc_acm.rst

```diff
@@ -0,0 +1,107 @@
+
+* Data transfer is only possible with the start of the USB device stack,
+  until then any data is discarded
+* If device is connected to the host, it still needs an application
+  on the host side which requests the data
+
+Devicetree compatible propertiy for CDC ACM UART is ``"zephyr,cdc-acm-uart"``.
+CDC ACM support is automatically selected when USB device support is enabled
+and a compatible node in devicetree sources is present. If necessary,
+CDC ACM support can be explicitly disabled by :kconfig:`CONFIG_USB_CDC_ACM`.
+About four CDC ACM UART instances can be defined and used,
+limited by the maximum number of supported endpoints on the controller.
+
+CDC ACM UART node is supposed to be child of a USB device controller node.
+Since the designation of the controller nodes varies from vendor to vendor,
+and our samples and application should be as generic as possible,
+the default USB device controller is usually assigned an ``zephyr_udc0``
+nodelabel. Often, CDC ACM UART is described in a devicetree overlay file
+and looks like this:
+
+.. code-block:: devicetree
+
+   &zephyr_udc0 {
+   	cdc_acm_uart0: cdc_acm_uart0 {
+   		compatible = "zephyr,cdc-acm-uart";
+   		label = "CDC_ACM_0";
+   	};
+   };
+
+Samples :ref:`usb_cdc-acm` and :ref:`usb_hid-cdc` have similar overlay files.
+And since no special properties are present, it may seem exaggerated to use
+devicetree to describe CDC ACM UART.  The motivation behind using devicetree
+is the easy interchangeability of real UART controller and CDC ACM UART
+in applications.
+
+Console over CDC ACM UART
+*************************
+
+With CDC ACM UART node from above and ``zephyr,console`` property of the
+chosen node, we can describe that CDC ACM UART is to be used with console.
+Similar overlay file is used by :ref:`cdc-acm-console`.
+If USB device support is enabled in the application, as in the console sampe,
+:kconfig:`CONFIG_USB_UART_CONSOLE` must be enabled,
+which does nothing but change the initialization time of the console driver.
+
+.. code-block:: devicetree
+
+   / {
+   	chosen {
+   		zephyr,console = &cdc_acm_uart0;
+   	};
+   };
+
+   &zephyr_udc0 {
+   	cdc_acm_uart0: cdc_acm_uart0 {
+   		compatible = "zephyr,cdc-acm-uart";
+   		label = "CDC_ACM_0";
+   	};
+   };
+
+Before the application uses the console, it is recommended to wait for
+the DTR signal:
+
+.. code-block:: c
+
+    const struct device *dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_console));
+    uint32_t dtr = 0;
+
+    if (usb_enable(NULL)) {
+    	return;
+    }
+
+    while (!dtr) {
+    	uart_line_ctrl_get(dev, UART_LINE_CTRL_DTR, &dtr);
+    	k_sleep(K_MSEC(100));
+    }
+
+    printk("nuqneH\n");
+
+CDC ACM UART as backend
+***********************
+
+As for the console sample, it is possible to configure CDC ACM UART as
+backend for other subsystems by setting :ref:`devicetree-chosen-nodes`
+properties.
+
+List of few Zephyr specific chosen properties which can be used to select
```

Comment:
What is your plan for maintaining this in the future? I do my best to keep the official list in doc/reference/devicetree/api.rst up to date, so I'm concerned about this going stale.

Reply:
I would like to keep it here as well but limited to the important use cases with the links to the examples.